src/hotspot/share/gc/parallel/objectStartArray.hpp

```diff
@@ -165,9 +165,11 @@ class ObjectStartArray : public CHeapObj<mtGC> {
     return *block != clean_block;
   }
 
-  // Return true if an object starts in the range of heap addresses.
-  // If an object starts at an address corresponding to
-  // "start", the method will return true.
+  // Return true iff an object starts in
+  //   [start_addr_aligned_down, end_addr_aligned_up)
+  // where
+  //   start_addr_aligned_down = align_down(start_addr, _card_size)
+  //   end_addr_aligned_up = align_up(end_addr, _card_size)
```

Comment:
The code doesn't do align_down on start, but looking at the callers it looks like start_addr will always be card size aligned. Would it make sense to assert this?

Reply:
`block_for_addr` does `align_down` implicitly. It's true that all callers provide aligned `start_addr`, but it feels odd that only one end is asserted aligned. Besides, this method doesn't really rely (or depend) on args being aligned.